OpenRA.Mods.Common/Traits/Upgrades/DeployToUpgrade.cs

```diff
@@ -60,14 +60,21 @@ public class DeployToUpgrade : IResolveOrder, IIssueOrder
 
 		bool isUpgraded;
 
-		public DeployToUpgrade(Actor self, DeployToUpgradeInfo info)
+		public DeployToUpgrade(ActorInitializer init, DeployToUpgradeInfo info)
 		{
-			this.self = self;
+			this.self = init.Self;
 			this.info = info;
 			manager = self.Trait<UpgradeManager>();
 			checkTerrainType = info.AllowedTerrainTypes.Count > 0;
 			canTurn = self.Info.HasTraitInfo<IFacingInfo>();
 			body = Exts.Lazy(self.TraitOrDefault<ISpriteBody>);
+			isUpgraded = init.Contains<DeployedInit>();
+		}
+
+		public void Created(Actor self)
+		{
+			if (isUpgraded)
+				GrantUpgrades();
```

Comment:
What makes you say that? It is set above by checking for `DeployedInit` in the `ActorInitializer`. This works.